src/DataLayouts/DataLayouts.jl

```diff
@@ -1253,6 +1253,21 @@ end
     dataview = @inbounds view(parent(data), v, Ni * z2 .+ (1:Ni))
     return dataview
 end
+# An iterator for datalayouts
+function Base.iterate(data::DataLayouts.AbstractData{S}, ivar = 1) where {S}
```

Comment:
If this isn't being used, perhaps remove it? I'm not sure that makes the most sense as a way to iterate.